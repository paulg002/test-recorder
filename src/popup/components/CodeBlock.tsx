import * as React from 'react';

export interface CodeBlockProps {
  index: number,
  text: string,
  dragStatus: string,
  destroyBlock: (i: number) => void,
  onDragStart: (e: React.DragEvent, i: number) => void,
  onDragOver: (e: React.DragEvent, i: number) => void,
  onDragEnd: () => void,
  onDrop: (e: React.DragEvent, i: number) => void,
}

export default ({
  index,
  text,
  destroyBlock,
  onDragStart,
  onDragOver,
  onDragEnd,
  onDrop,
  dragStatus,
}: CodeBlockProps) => {
  const i = text.startsWith('cy.url()')
    ? text.lastIndexOf('\'', text.length - 4)
    : text.indexOf('\'');
  const j = text.lastIndexOf('\'') + 1;
  const preSelector = text.slice(0, i);
  const selector = text.slice(i, j);
  const postSelector = text.slice(j);
  return (
    <li
      className={dragStatus}
      draggable
      onDragStart={e => onDragStart(e, index)}
      onDragEnd={onDragEnd}
      onDragOver={e => onDragOver(e, index)}
      onDrop={e => onDrop(e, index)}
    >
      <span>
        {preSelector}
        <mark className="selector">
          {selector}
        </mark>
        {postSelector}
      </span>
      <button type="button" className="delete" onClick={() => destroyBlock(index)}>x</button>
    </li>
  );
};
