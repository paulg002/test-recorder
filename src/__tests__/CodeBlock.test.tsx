import * as React from 'react';
import { shallow } from 'enzyme';
import CodeBlock, { CodeBlockProps } from '../popup/components/CodeBlock';
import '../setupTests';

describe('CodeBlock', () => {
  let wrapper;
  const props: CodeBlockProps = {
    index: 0,
    text: 'cy.get(\'.button\').click();',
    dragStatus: 'code-block',
    destroyBlock: jest.fn(),
    onDragStart: jest.fn(),
    onDragOver: jest.fn(),
    onDragEnd: jest.fn(),
    onDrop: jest.fn(),
  };
  describe('CodeBlock', () => {
    it('Should render a CodeBlock', () => {
      wrapper = shallow(<CodeBlock {...props} />);
      expect(wrapper).toMatchSnapshot();
      const item = wrapper.find('li');
      expect(item.hasClass(props.dragStatus)).toBe(true);
      expect(item.find('mark').hasClass('selector')).toBe(true);
    });
  });
});
