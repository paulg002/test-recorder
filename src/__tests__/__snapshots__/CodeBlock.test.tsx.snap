// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeBlock CodeBlock Should render a CodeBlock 1`] = `
<li
  className="code-block"
  draggable={true}
  onDragEnd={[MockFunction]}
  onDragOver={[Function]}
  onDragStart={[Function]}
  onDrop={[Function]}
>
  <span>
    cy.get(
    <mark
      className="selector"
    >
      '.button'
    </mark>
    ).click();
  </span>
  <button
    className="delete"
    onClick={[Function]}
    type="button"
  >
    x
  </button>
</li>
`;
